src/module/actor/SR5Actor.ts

```diff
@@ -199,178 +218,212 @@ export class SR5Actor extends Actor {
 
-        // CALCULATE RECOIL
-        data.recoil_compensation = 1 + Math.ceil(attributes.strength.value / 3);
+        // tracks
+        for (let [t, tr] of Object.entries(track)) {
+            tr.label = CONFIG.SR5.damageTypes[t];
+        }
+    }
+
+    /**
+     * Prepare actor data movement
+     * @param data
+     */
+    static prepareMovement(data: SR5ActorData) {
+        const { attributes, modifiers } = data;
+        const movement = data.movement;
+        // default movement: WALK = AGI * 2, RUN = AGI * 4
+        movement.walk.value = attributes.agility.value * (2 + Number(modifiers['walk']));
+        movement.run.value = attributes.agility.value * (4 + Number(modifiers['run']));
+    }
+
+    /**
+     * Prepare the modifiers that are displayed in the Misc. tab
+     * @param data
+     */
+    static prepareModifiers(data: SR5ActorData) {
+        if (!data.modifiers) data.modifiers = {};
+        const modifiers = {};
+        let miscTabModifiers = [
+            'soak',
+            'drain',
+            'armor',
+            'physical_limit',
+            'social_limit',
+            'mental_limit',
+            'stun_track',
+            'physical_track',
+            'meat_initiative',
+            'meat_initiative_dice',
+            'astral_initiative',
+            'astral_initiative_dice',
+            'matrix_initiative',
+            'matrix_initiative_dice',
+            'composure',
+            'lift_carry',
+            'judge_intentions',
+            'memory',
+            'walk',
+            'run',
+            'defense',
+            'wound_tolerance',
+            'essence',
+            'fade',
+        ];
+        miscTabModifiers.sort();
+        // force global to the top
+        miscTabModifiers.unshift('global');
+
+        for (let item of miscTabModifiers) {
+            modifiers[item] = Number(data.modifiers[item]) || 0;
+        }
+
+        data.modifiers = modifiers;
+    }
 
-        // INITIATIVE
-        const init = data.initiative;
-        init.meatspace.base.base = attributes.intuition.value + attributes.reaction.value + modifiers['meat_initiative'];
-        init.meatspace.dice.base = 1 + modifiers['meat_initiative_dice'];
-        init.astral.base.base = attributes.intuition.value * 2 + modifiers['astral_initiative'];
-        init.astral.dice.base = 2 + modifiers['astral_initiative_dice'];
-        init.matrix.base.base = attributes.intuition.value + data.matrix.data_processing.value + modifiers['matrix_initiative'];
-        init.matrix.dice.base = data.matrix.hot_sim ? 4 : 3 + modifiers['matrix_initiative_dice'];
-        if (init.perception === 'matrix') init.current = init.matrix;
-        else if (init.perception === 'astral') init.current = init.astral;
+    /**
+     * Prepare actor data for initiative
+     * @param data
+     */
+    static prepareInitiative(data: SR5ActorData) {
```

Comment:
I don't like how this is only effecting and being called from an instance but is a static method. You're operating on instance data from a static context, which defeats the purpose of an static method.